client/src/templates/Challenges/components/preview-portal.tsx

```diff
@@ -51,37 +91,63 @@ class PreviewPortal extends Component<PreviewPortalProps> {
       }
     `;
 
-    this.externalWindow = window.open(
-      '',
-      '',
-      'width=960,height=540,left=100,top=100'
-    );
-
-    this.externalWindow?.document.head.appendChild(this.titleEl);
-    this.externalWindow?.document.head.appendChild(this.styleEl);
-    this.externalWindow?.document.body.setAttribute(
-      'style',
-      `
+    const processWindowAndDocuments = (
+      title: HTMLElement,
+      style: HTMLElement,
+      containerEl: HTMLElement,
+      externalWindow: Window | null,
+      clientDocument: Document | undefined
+    ) => {
+      externalWindow?.document.head.appendChild(title);
+      externalWindow?.document.head.appendChild(style);
+      externalWindow?.document.body.setAttribute(
+        'style',
+        `
         margin: 0px;
         padding: 0px;
         overflow: hidden;
       `
-    );
-    this.externalWindow?.document.body.appendChild(this.containerEl);
+      );
+      externalWindow?.document.body.appendChild(containerEl);
+      storePortalWindow(externalWindow);
+      storeportalDocument(clientDocument);
+    };
+
+    if (portalWindow && showPreviewPortal && portalDocument) {
+      this.existingExternalWindow.document.head.innerHTML = '';
+      this.existingExternalWindow.document.body.innerHTML = '';
+      processWindowAndDocuments(
+        this.titleEl,
+        this.styleEl,
+        this.containerEl,
+        this.existingExternalWindow,
+        this.existingExternalWindow?.document
```

Comment:
If the document is always `externalWindow.document`, do we still need to store it in the redux store? Isn't storing just the window enough?

Reply:
disreguard this code. It is not good. I will make a new pull request you will see it shortly.

Reply:
@nicolo-ribaudo check the new PR on the issue it is much better. I think you will like it.